src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((result, item) => {
+    const keyValue = item[key];
```

Comment:
Ensure that the 'key' property exists on 'item' before accessing it to avoid potential runtime errors. You can use a type guard to check if the property exists.